firmware/user/modes/mode_joust_game.c

```diff
@@ -0,0 +1,1006 @@
+ * a successful connection. After the animation, the game will start
+ *
+ * @param arg unused
+ */
+void ICACHE_FLASH_ATTR joustGameLedTimeout(void* arg __attribute__((unused)) )
+{
+    switch(joust.led.ConnLedState)
+    {
+        case LED_CONNECTED_BRIGHT:
+        {
+            joust.led.currBrightness++;
+            if(joust.led.currBrightness == 30)
+            {
+                joust.led.ConnLedState = LED_CONNECTED_DIM;
+            }
+            break;
+        }
+        case LED_CONNECTED_DIM:
+        {
+            joust.led.currBrightness--;
+            if(joust.led.currBrightness == 0)
+            {
+                joust.led.ConnLedState = LED_CONNECTED_BRIGHT;
+            }
+            break;
+        }
+        case LED_OFF:
+        case LED_ON_1:
+        case LED_DIM_1:
+        case LED_ON_2:
+        case LED_DIM_2:
+        case LED_OFF_WAIT:
+        default:
+        {
+            // No other cases handled
+            break;
+        }
+    }
+    uint8_t i;
+    for(i = 0; i < 6; i++)
+    {
+        joust.led.Leds[i].r = (EHSVtoHEX(joust.con_color, 255,  joust.led.currBrightness) >>  0) & 0xFF;
+        joust.led.Leds[i].g = (EHSVtoHEX(joust.con_color, 255,  joust.led.currBrightness) >>  8) & 0xFF;
+        joust.led.Leds[i].b = (EHSVtoHEX(joust.con_color, 255,  joust.led.currBrightness) >> 16) & 0xFF;
+    }
+
+    setLeds(joust.led.Leds, sizeof(joust.led.Leds));
+}
+
+/**
+ * This is called after connection is all done. Start the game!
+ *
+ * @param arg unused
+ */
+void ICACHE_FLASH_ATTR joustStartPlaying(void* arg __attribute__((unused)))
+{
+    joust_printf("%s\r\n", __func__);
+    joust_printf("\nstarting the game\n");
+
+    // Disable button debounce for minimum latency
+    enableDebounce(false);
+
+    // Turn off the LEDs
+    joustDisarmAllLedTimers();
+    joust.led.currBrightness = 0;
+    joust.led.ConnLedState = LED_CONNECTED_BRIGHT;
+    os_timer_arm(&joust.tmr.GameLed, 6, true);
+    joust.gameState = R_PLAYING;
+
+}
+
+/**
+ * Start a round of the game by picking a random action and starting
+ * refGameLedTimeout()
+ */
+void ICACHE_FLASH_ATTR joustStartRound(void)
+{
+    joust.gameState = R_PLAYING;
+
+}
+
+void ICACHE_FLASH_ATTR joustUpdateDisplay(void)
+{
+    // Clear the display
+    clearDisplay();
+    // Draw a title
+    plotText(0, 0, "JOUST", RADIOSTARS);
+    // Display the acceleration on the display
+    char accelStr[32] = {0};
+
+    ets_snprintf(accelStr, sizeof(accelStr), "Acc: %d", joust.rolling_average);
+    plotText(0, OLED_HEIGHT - (1 * (FONT_HEIGHT_IBMVGA8 + 1)), accelStr, IBM_VGA_8);
+}
+
+/**
+ * Update the acceleration for the Joust mode
+ */
+void ICACHE_FLASH_ATTR joustAccelerometerHandler(accel_t* accel)
+{
+
+    joust.joustAccel.x = accel->x;
+    joust.joustAccel.y = accel->y;
+    joust.joustAccel.z = accel->z;
+    int mov = (int) sqrt(pow(joust.joustAccel.x, 2) + pow(joust.joustAccel.y, 2) + pow(joust.joustAccel.z, 2));
```

Comment:
for some reason having the sqrt() function stabilizes the sensitivity much more than just the sum of the powers, I tested the other way as well. I think it might be best to leave it in.